pkg/operator/ceph/pool/controller.go

```diff
@@ -137,13 +137,33 @@ func poolChanged(old, new cephv1.PoolSpec) bool {
 	return false
 }
 
+func (c *PoolController) handlePoolDelete(name, namespace string) error {
```

Comment:
Currently this check will only apply to the deletes of the pool CRD. Let's move all these checks for images and snapshots in the pool into the common [DeletePool()](https://github.com/rook/rook/blob/87ca2086406674d64f6129d849bfa7764205f155/pkg/daemon/ceph/client/pool.go#L155) method. This way the same checks will apply whenever any pool is deleted, whether from the pool CRD, pools from the object store, or pools from the filesystem.

Reply:
done, added a test case  for the same